pkg/pillar/zedcloud/authen.go

```diff
@@ -559,6 +576,43 @@ func verifySignature(log *base.LogObject, certByte []byte, interm *x509.CertPool
 		log.Errorln("verifySignature: " + errStr)
 		return errors.New(errStr)
 	}
+
+	// verify cert validity dates against thresholds
```

Comment:
I think it is both simpler and more correct to do not this here, but instead  add a "when" argument to verifySignature() and call verifySignature() with different values of when (not+30 days etc)

And the feed what when into x509.VerifyOptions so that the leafcert.Verify above will do all the checks.

Reply:
Sounds better

Reply:
One question, should we return a failure, or just log on the threshold events?

Reply:
For now let's get the logging in place (with the periodic check) and in a future PR once we've gone through a certificate roll in production, we can add the hard fail.

In terms of placement of the periodic check, after you started this work we've made controllerCertsTask() periodically try to re-fetch the certificates. It probably makes sense having that function also re-verify the validity of the certificates both each time when it fetched new ones, and infrequently (once a day?) when it did not fetch new certificates.